plugins/main/windows/l2bridge/l2bridge_windows.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2017 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"runtime"
+	"strings"
+
+	"github.com/juju/errors"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+	"github.com/containernetworking/plugins/pkg/hns"
+	"github.com/containernetworking/plugins/pkg/ipam"
+)
+
+type NetConf struct {
+	hns.NetConf
+
+	IPMasq               bool   `json:"ipMasq"`
+	ClusterNetworkPrefix string `json:"clusterNetworkPrefix,omitempty"`
+}
+type K8sCniEnvArgs struct {
+	types.CommonArgs
+
+	K8S_POD_NAMESPACE types.UnmarshallableString `json:"K8S_POD_NAMESPACE,omitempty"`
+	// K8S_POD_NAME             types.UnmarshallableString `json:"K8S_POD_NAME,omitempty"`
+	// K8S_POD_INFRA_CONTAINER_ID types.UnmarshallableString `json:"K8S_POD_INFRA_CONTAINER_ID,omitempty"`
+}
```

Comment:
This is Kubernetes specific, which isn't something we want in CNI since CNI is not Kubernetes-specific and works with other runtimes too. For Kube I believe there are some other solutions that we can explore that don't hardcode Kube-ness into the CNI plugin.

For now, let's remove all the parseCniArgs() and K8S_POD_NAMESPACE stuff and fix that in a follow-up patch.